packages/dart_firebase_admin/lib/src/google_cloud_firestore/timestamp.dart

```diff
@@ -81,8 +91,29 @@ final class Timestamp implements _Serializable {
   /// as the given number of milliseconds.
   factory Timestamp.fromMillis(int milliseconds) {
     final seconds = (milliseconds / 1000).floor();
-    final nanos = (milliseconds - seconds * 1000) * _msToNanos;
-    return Timestamp._(seconds: seconds, nanoseconds: nanos);
+    final nanos = (milliseconds - seconds * 1000 * 1000) * _msToNanos;
+
+    return Timestamp(seconds: seconds, nanoseconds: nanos);
+  }
+
+  /// Creates a new timestamp from the given number of microseconds.
+  ///
+  /// ```dart
+  /// final documentRef = firestore.doc('col/doc');
+  ///
+  /// documentRef.set({ 'startTime': Timestamp.fromMicros(42) });
+  /// ```
+  ///
+  /// - [microseconds]: Number of microseconds since Unix epoch
+  /// 1970-01-01T00:00:00Z.
+  ///
+  /// Returns a new [Timestamp] representing the same point in time
+  /// as the given number of microseconds.
+  factory Timestamp.fromMicros(int microseconds) {
```

Comment:
This one needs a test too

Reply:
Done! Thanks : )

Reply:
Is this constructor necessary? It makes sense to stick as close to the JavaScript implementation as possible imo, and I don't believe this exists there (could be wrong)

Reply:
I believe the Timestamp.fromMicros constructor is necessary. Considering the Dart native environment, the lack of this constructor would result in the loss of microsecond precision.

Additionally, while the microsecondsSinceEpoch property of the Dart DateTime API is available in the web environment, the limitations of JavaScript mean that there will still be a loss of precision in the microsecond unit.

Even though a code like 
```dart
Timestamp.fromMicros(DateTime.now().microsecondsSinceEpoch);
```
would not throw an error

Reply:
Interesting, thanks for the insight.  I'd much prefer the name to be `fromMicroseconds`, but leave that one to Remi to call :D